### ISSUE TITLE
Packer cannot authenticate

### ISSUE THREAD

**Issue by jszakmeister** [hashicorp/packer#7475]
When trying to build a Fedora 29 VM using VMware Fusion 11 and the vmware-iso builder, I keep running into errors when it attempts to ssh into the VM:

> 2019/04/07 05:41:34 packer: 2019/04/07 05:41:34 [DEBUG] SSH handshake err: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none password], no supported methods remain

I'm running Packer v1.3.5 on macOS High Sierra (10.13.6).  I've run with `-on-error=abort` so that I could check ssh'ing into the VM myself and it was successful.  From the packer logs I can see that it was pulling the correct ip address too (though the "trying to connect" line in the normal output often shows the wrong ip).  The packer.log file is [here](https://gist.github.com/jszakmeister/919a82053c51c85243348266a3be6264#file-packer-log).  The gist containing everything to reproduce this is [here](https://gist.github.com/jszakmeister/919a82053c51c85243348266a3be6264).

Note: to rule out an ssh configuration error on my end, I made sure no agents were running and moved my ssh config out of the way, but it didn't help.  Unfortunately, there wasn't enough in the logs for me.  I've included the [sshd_config](https://gist.github.com/jszakmeister/02b6710a20917533eef62330f3e03c75) from the vm here, just in case you need it.

I'm not sure how to build packer (I've only dabbled in Go, I've never really used it), but I'm willing to try some things if you think it would help.  For what it's worth, I was able to build a CentOS 7 VM without issue, so it smells like some feature is unsupported or something new appeared in Fedora 29 that packer isn't ready for.  That said, there's also evidence that it used to work as several people have packer recipes out there for the vmware-iso builder.  I haven't tried building one of those just yet, but as far as I can tell, there appears to be a bug in here somewhere.

**Comment by rickard-von-essen** [comment]
Could you provide the content of `journalctl -r /usr/sbin/sshd` when packer is trying to connect? The sshd config including defaults from `sshd -T`. 

And when you manually connect with the ssh client add `-vvv` and provide that to.o.

**Comment by jszakmeister** [comment]
Interesting.  `journalctl -r /usr/sbin/sshd` is showing a bunch of repeated entries like this:

```text
Apr 07 20:41:16 build1 sshd[1073]: Connection closed by invalid user server 192.168.64.1 port 49796 [preauth]
Apr 07 20:41:16 build1 sshd[1073]: Failed password for invalid user server from 192.168.64.1 port 49796 ssh2
Apr 07 20:41:14 build1 sshd[1073]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.>
Apr 07 20:41:14 build1 sshd[1073]: pam_unix(sshd:auth): check pass; user unknown
Apr 07 20:41:14 build1 sshd[1073]: Invalid user server from 192.168.64.1 port 49796
Apr 07 20:41:14 build1 sshd[1072]: Did not receive identification string from 192.168.64.1 port 49795
```

Shoot... it's trying to log in as the wrong user.  I had `ssh_username` and `ssh_password` backwards--the password as the username, and vice versa.  Arrgh.  I'm sorry I bother you guys with this.  I spent a ton of hours trying to get it figured out and not once did I look at the logs for sshd.  Thank you for the suggestion.
